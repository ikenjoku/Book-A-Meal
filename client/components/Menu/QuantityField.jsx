import React, { Component } from 'react';
import PropTypes from 'prop-types';

const QuantityField = ({ quantity, handleChange }) => {
  return (
    <input
      className="quantity-field"
      type="number"
      name="quantity"
      min="1"
      value={quantity}
      onChange={handleChange}
    />
  );
};

export default QuantityField;
