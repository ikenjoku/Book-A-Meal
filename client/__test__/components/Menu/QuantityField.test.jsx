import React from 'react';
import { shallow } from 'enzyme';
import QuantityField from '../../../components/Menu/QuantityField';

describe('QuantityField Component', () => {
  let wrapper;
  const quantity = 1;
  const handleChange = jest.fn();

  test('should render QuantityField correctly', () => {
    wrapper = shallow(<QuantityField
      quantity={quantity}
      handleChange={handleChange}
    />);
    expect(wrapper).toMatchSnapshot();
  });
});
