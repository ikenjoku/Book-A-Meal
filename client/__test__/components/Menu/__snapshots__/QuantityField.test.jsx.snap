// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuantityField Component should render QuantityField correctly 1`] = `
<input
  className="quantity-field"
  min="1"
  name="quantity"
  onChange={[MockFunction]}
  type="number"
  value={1}
/>
`;
